### ISSUE TITLE
Overloaded decorator using TypeVarTuple gets reported as not consistent with overloads.

### ISSUE THREAD

**Issue by ImogenBits** [microsoft/pyright#7502]
I want to implement a decorator with the common pattern of it taking some optional parameters and then being usable either directly or with a function call on it, like this:
```py
@my_wrapper
def f(): ...

@my_wrapper(first=1, second=2, third=3)
def g(): ...
```
Usually this can be easily done via overloads and they also typecheck correctly with pyright. But when I use a TypeVarTuple in the type of the decorated and returned function types it reports it as not being compatible with the overloads. Here's a simplified example snippet:
```py
from collections.abc import Callable
from typing import TypeVarTuple, overload

Args = TypeVarTuple("Args")
MyFunc = Callable[[*Args], int]
ApplyDecorator = Callable[[MyFunc[*Args]], MyFunc[*Args]]


@overload
def my_wrapper(function: MyFunc[*Args]) -> MyFunc[*Args]: ...
@overload
def my_wrapper() -> ApplyDecorator[*Args]: ...

def my_wrapper(function: MyFunc[*Args] | None = None) -> MyFunc[*Args] | ApplyDecorator[*Args]: ...
```
With this error message:
```
Overloaded implementation is not consistent with signature of overload 2
  Function return type "ApplyDecorator[*Args@my_wrapper]" is incompatible with type "MyFunc[*Args@my_wrapper] | ApplyDecorator[*Args@my_wrapper]"
    Type "ApplyDecorator[*Args@my_wrapper]" cannot be assigned to type "MyFunc[*Args@my_wrapper] | ApplyDecorator[*Args@my_wrapper]"
      Type "ApplyDecorator[*Args@my_wrapper]" cannot be assigned to type "MyFunc[*Args@my_wrapper]"
        Parameter 1: type "*Args@my_wrapper" cannot be assigned to type "*tuple[(*Args@my_wrapper) -> int]"
          Type "*Args@my_wrapper" cannot be assigned to type "*tuple[(*Args@my_wrapper) -> int]"
        Function return type "(*Args@my_wrapper) -> int" is incompatible with type "int"
          "function" is incompatible with "int"
      Type "ApplyDecorator[*Args@my_wrapper]" cannot be assigned to type "ApplyDecorator[*Args@my_wrapper]"
```

When I replace the TypeVarTuple with either a normal TypeVar or a concrete type the decorator type checks as expected. If this actually is the expected behavior type var tuples, what is the correct way to use them here?

I am using the VSCode Pylance language server version 2024.3.1 with pyright version 1.1.351.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.355, which I just published.
